src/nwsapi.js

```diff
@@ -810,7 +810,7 @@
           // tag name resolver
           case (/[_a-z]/i.test(symbol) ? symbol : undefined):
             match = selector.match(Patterns.tagName);
-            source = 'if((e.localName=="' + match[1] + '")){' + source + '}';
+            source = 'if((e.localName=="' + match[1].toLowerCase() + '")){' + source + '}';
```

Comment:
## Before

<img width="1789" alt="Screenshot 2024-02-06 at 4 34 16 PM" src="https://github.com/dperini/nwsapi/assets/716969/ffc8a8ec-74e1-4e3f-8a5c-9625353bfaa3">


## After

<img width="1789" alt="Screenshot 2024-02-06 at 4 34 45 PM" src="https://github.com/dperini/nwsapi/assets/716969/02a0149a-a3b2-4258-a1e7-af73752de194">

Reply:
Hi @Manvel and thank you for your contribution to "nwsapi".
If you can prepare a test were this fails before applying your change it would help.
There is only another environment were we must make sure this doesn't create regressions.
This is when developers wants to have mixed contents (HTML/XML) in the same page but I have tests for that.
If it passes on those cases I will apply it asap, possibly in upcoming 2.8 release.

Reply:
Thanks for the reply @dperini! My pleasure, always happy to support open source when/if I can.

From what I see the report is same on master and current branch:

## svg-test.html

<img width="643" alt="Screenshot 2024-02-13 at 11 43 26 AM" src="https://github.com/dperini/nwsapi/assets/716969/c3381375-441b-4661-8aa7-516427d7ec9e">

`hasAttributeNS` seem to be missing from the Dom API.

## xml-test.html

<img width="634" alt="Screenshot 2024-02-13 at 11 43 55 AM" src="https://github.com/dperini/nwsapi/assets/716969/79ef366d-1812-49d7-8e8d-73c68ae1142e">

Maybe at some point a regression was introduced?

## Case sensitivity

As for current change there does not seem to be a test case covering cases, [XML suppose to be case sensitive while HTML is case insensitive](https://www.w3.org/TR/selectors-4/#case-sensitive), so they might need to have different approaches when it comes to case sensitivity.


## Actions

- I will try having closer look to figure out when the regressions has been introduced for SVG and XML tests and try fixing them. 
- Will add another test case for case sensitivity, that would probably require some changes in IMPL, right now there seem to be no consideration about that, so think with or without those change we are not making situation worse, considering the wide use of current library in HTML, think it would make sense to make it compatible there.

## Note

I will be busy with team offsite for next 11 days, maybe slow replying/implementing, but will ensure to address the comment and try finishing the PR.

Reply:
> If you can prepare a test were this fails before applying your change it would help.

Have splitted the commits, so test comes first now. If that what you were asking for, otherwise please let me know.

> There is only another environment were we must make sure this doesn't create regressions.
This is when developers wants to have mixed contents (HTML/XML) in the same page but I have tests for that.
> If it passes on those cases I will apply it asap, possibly in upcoming 2.8 release.

I assume you are referring to `test/xml/xml-test.html` and `test/xml/svg-test.html`, right?

Those tests seem to be broken on master [as mentioned above](https://github.com/dperini/nwsapi/pull/101/files#r1487627109). @dperini  If you were referring to different tests please let me know. Not yet sure how they need to be fixed.

Bisect suggests that the regression for `test/xml/xml-test.html` was introduced by: https://github.com/dperini/nwsapi/commit/87fbfa9e3974571edc192689b0712d9536839246 commit.

Reply:
@dperini I am currently stuck, you think we can land the changes for HTML and fix the XML regressions for later on? As the XML tests seem to have been regressed quite a while ago(Unless I am missing smth). Please let me know what you think what will be sensible here.